UnityProject/Assets/Scripts/Transform/CustomNetTransform.cs

```diff
@@ -542,6 +545,22 @@ private void AdminRespawn()
 	{
 		PlayerManager.PlayerScript.playerNetworkActions.CmdAdminRespawn(gameObject);
 	}
+
+	// Checks if the object is occupiable and update occupant position if it's occupied (ex: a chair)
+	private void UpdateOccupant()
+	{
+		// Since we are at the transform level, we go up one level to get the parent
+		OccupiableDirectionalSprite occupiableDirectionalSprite = GetComponentInParent<OccupiableDirectionalSprite>();
```

Comment:
For performance reasons it's best to avoid GetComponent in update loops, which I THINK this might occur in. It creates garbage.

Instead, you should just call `GetComponent<OccupiableDirectionalSprite>()` and save it to a private field in CustomNetTransform's Start method. Then in this method you can check if the field is null to decide if this needs to update its occupant.